sklearn/utils/tests/test_testing.py

```diff
@@ -491,3 +493,147 @@ def test_check_docstring_parameters():
          'type definition for param: "c " (type definition was "")',
          'sklearn.utils.tests.test_testing.f_check_param_definition There was '
          'no space between the param name and colon ("d:int")'])
+
+
+def func_doc1(y_true, y_pred, sample_weight):
+    """Dummy function for docstring testing.
+
+    Parameters
+    ----------
+    y_true : 1d array-like
+        Ground truth (correct) target values.
+
+    y_pred : 1d array-like
+        Estimated targets as returned by a classifier.
+
+    sample_weight : array-like of shape = [n_samples], optional
+        Sample weights.
+
+    Returns
+    -------
+    fbeta_score : float or array of float, shape = [n_unique_labels]
+        F-beta score of the positive class in binary classification.
+
+    precision : float or array of float, shape = [n_unique_labels]
+        Precision values.
+
+    recall : float or array of float, shape = [n_unique_labels]
+
+    """
+    pass
+
+
+def func_doc2(y_true, y_pred, sample_weight):
+    """Dummy function for docstring testing.
+
+    Parameters
+    ----------
+    y_true : 1d array-like
+        Ground truth (correct) target values.
+
+    y_pred : 1d array-like
+        Estimated targets as returned by a classifier.
+
+    sample_weight : array-like of shape = [n_samples], optional
+
+    Returns
+    -------
+    fbeta_score : float or array of float, shape = [n_unique_labels]
+        F-beta score of the positive class in binary classification.
+
+    precision : float or array of float
+        Precision values.
+
+    """
+    pass
+
+
+@if_numpydoc
+def test_assert_consistent_docs():
+    # Testing invalid object type
+    assert_raises(TypeError, assert_consistent_docs, ["Object1", "Object2"])
+
+    # Testing with dummy functions
+    assert_consistent_docs([func_doc1, func_doc2],
+                           include_params=['y_true', 'y_pred'],
+                           include_returns=False)
+
+    from numpydoc import docscrape  # using NumpyDocString object for tests
+
+    doc1 = docscrape.NumpyDocString(inspect.getdoc(func_doc1))
+    doc2 = docscrape.NumpyDocString(inspect.getdoc(func_doc2))
+
+    # Test error messages on mismatch
+    error_msg1 = ("Parameter 'sample_weight' of 'Object 2' has inconsistent"
+                  " description with that of 'Object 1'.")
+    assert_raise_message(AssertionError, error_msg1, assert_consistent_docs,
+                         [doc1, doc2], include_params=['sample_weight'],
+                         include_returns=False)  # description mismatch
+
+    error_msg2 = ("Return 'precision' of 'Object 2' has inconsistent type"
+                  " definition with that of 'Object 1'.")
+    assert_raise_message(AssertionError, error_msg2, assert_consistent_docs,
+                         [doc1, doc2], include_returns=['precision'],
+                         include_params=False)  # type definition mismatch
+
+    doc3 = doc1  # both doc1 and doc3 return 'recall' whereas doc2 does not
```

Comment:
a thought inspired by this: I wonder if we should raise an error/warning if an explicitly included name is only in one of the input docstrings...

Reply:
I am not sure it would be very necessary. What if a name is present only in a few of the objects, maybe there should be a warning for that as well.